data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/IndexConfiguration.java

```diff
@@ -93,6 +87,28 @@ private IndexConfiguration(final Builder builder) {
         this.ismPolicyFile = builder.ismPolicyFile;
     }
 
+    private void determineIndexType(Builder builder) {
+        if(builder.indexType != null) {
+            Optional<IndexType> mappedIndexType = IndexType.getByValue(builder.indexType);
+            mappedIndexType.ifPresent(type -> {indexType = type;});
```

Comment:
This is not a common pattern I've seen with Optionals and lambdas.

A more idiomatic way would be to combine this line with the one below as:

```
indexType = mappedIndexType.orElseThrow(...);
```

Reply:
Great suggestion! I didn't know that. Optimized in new revision.